gcloud-java-dns/src/main/java/com/google/gcloud/spi/DefaultDnsRpc.java

```diff
@@ -0,0 +1,192 @@
+package com.google.gcloud.spi;
+
+import static com.google.gcloud.spi.DnsRpc.Option.DNS_NAME;
+import static com.google.gcloud.spi.DnsRpc.Option.DNS_TYPE;
+import static com.google.gcloud.spi.DnsRpc.Option.FIELDS;
+import static com.google.gcloud.spi.DnsRpc.Option.PAGE_SIZE;
+import static com.google.gcloud.spi.DnsRpc.Option.PAGE_TOKEN;
+import static com.google.gcloud.spi.DnsRpc.Option.SORTING_ORDER;
+import static java.net.HttpURLConnection.HTTP_NOT_FOUND;
+
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.services.dns.Dns;
+import com.google.api.services.dns.model.Change;
+import com.google.api.services.dns.model.ChangesListResponse;
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.ManagedZonesListResponse;
+import com.google.api.services.dns.model.Project;
+import com.google.api.services.dns.model.ResourceRecordSet;
+import com.google.api.services.dns.model.ResourceRecordSetsListResponse;
+import com.google.gcloud.dns.DnsException;
+import com.google.gcloud.dns.DnsOptions;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * A default implementation of the DnsRpc interface.
+ */
+public class DefaultDnsRpc implements DnsRpc {
+
+  private static final String SORTING_KEY = "changeSequence";
+  private final Dns dns;
+  private final DnsOptions options;
+
+  private static DnsException translate(IOException exception) {
+    return new DnsException(exception);
+  }
+
+  /**
+   * Constructs an instance of this rpc client with provided {@link DnsOptions}.
+   */
+  public DefaultDnsRpc(DnsOptions options) {
+    HttpTransport transport = options.httpTransportFactory().create();
+    HttpRequestInitializer initializer = options.httpRequestInitializer();
+    this.dns = new Dns.Builder(transport, new JacksonFactory(), initializer)
+        .setRootUrl(options.host())
+        .setApplicationName(options.applicationName())
+        .build();
+    this.options = options;
+  }
+
+  @Override
+  public ManagedZone create(ManagedZone zone) throws DnsException {
```

Comment:
I now wonder why we didn't provide options also for create (in the API and RPC layer)...